### PR TITLE
Stats: Rename existing highlight cards

### PR DESCRIPTION
**Pull request by jsnmoon**
Related to #72493

## Proposed Changes

* Renames "HighlightCards" to "WeeklyHighlightCards".
* Renames "HighlightCard" to "CountComparisonCard".

These changes are in preparation for a larger refactor for `CountComparisonCard` and `WeeklyHighlightCards`. It also establishes a naming convention for other card collections like `AllTimeHighlightCards` and `SubscriberHighlightCards` that currently exist outside of the components package.

## Testing Instructions

* Open the live branch for this PR.
* Ensure that the traffic and insights pages of Calypso Stats look and work as expected.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by a8ck3n** [comment]
This tests well for me. Not specific to this PR but I'm wondering if we should adjust the minimum width on these cards to prevent the following:

SCR-20230420-ovdb

My personal preference would be to prevent wrapping of the red/green status text.

File: client/my-sites/stats/annual-highlights-section/index.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { AnnualHighlightCards } from '@automattic/components';
+import AnnualHighlightCards from '@automattic/components/src/highlight-cards/annual-highlight-cards';
```
Comment (a8ck3n) [inline client/my-sites/stats/annual-highlights-section/index.tsx]: I'm curious about this change to direct imports. I see it was changed back in a later commit. Was this just to keep the build system happy while the renaming was happening?

File: packages/components/src/highlight-cards/annual-highlight-cards.tsx [hunk 2]
```diff
@@ -50,31 +50,31 @@ export default function AnnualHighlightCards( {
 			</div>
 
 			<div className="highlight-cards-list">
-				<HighlightCard
+				<CountComparisonCard
```
Comment (a8ck3n) [inline packages/components/src/highlight-cards/annual-highlight-cards.tsx:53]: Would it make sense to pull the `heading` and `icon` values up a level and provide them when we are setting up the `count` values?

Currently we transform the API data in the caller and provide the `counts` configuration object. If we instead generated an array of card data objects (heading, icon, count) in the caller, we could avoid having configuration spread across two files. We could then map the passed in array here so this component would no longer be concerned about the number of cards being displayed.

**Review by a8ck3n — APPROVED** [review]
One more comment. No blockers! Tests great!

**Comment by jsnmoon** [comment]
> I'm wondering if we should adjust the minimum width on these cards to prevent the following (@a8ck3n)

Sounds like a good idea to me!

File: client/my-sites/stats/annual-highlights-section/index.tsx (hunk 1 above)
Comment (jsnmoon) [reply client/my-sites/stats/annual-highlights-section/index.tsx]: I wanted to verify if our build system was properly tree-shaking unused dependencies for each entry point; I was concerned that exporting additional components in `src/index.tsx` may bloat all bundles importing the components package.

Luckily, our build system does properly tree-shake unused dependencies. I added them back to `index.tsx` to improve the DevUX for consuming these components.

File: packages/components/src/highlight-cards/annual-highlight-cards.tsx (hunk 2 above)
Comment (jsnmoon) [reply packages/components/src/highlight-cards/annual-highlight-cards.tsx:53]: Yes, I think that'd be a good next step for refactoring the highlight cards 🙂

File: client/my-sites/stats/annual-highlights-section/index.tsx (hunk 1 above)
Comment (a8ck3n) [reply client/my-sites/stats/annual-highlights-section/index.tsx]: Nice! Thanks for the clarification!

**Pull request merged**